File: JavaToKotlin/app/src/main/java/io/dustin/apps/board/domain/restriction/posting/repository/PostingRestrictionRepository.java
package io.dustin.apps.board.domain.restriction.posting.repository;

import io.dustin.apps.board.domain.restriction.posting.model.PostingRestriction;
import io.dustin.apps.common.repository.BaseRepository;

import java.util.Optional;

public interface PostingRestrictionRepository extends BaseRepository<PostingRestriction, Long> {

    void deleteByFromUserIdAndToUserIdAndPostingId(Long fromUserId, Long toUserId, Long postingId);
    Optional<PostingRestriction> findByFromUserIdAndToUserIdAndPostingId(Long fromUserId, Long toUserId, Long postingId);

}

File: apps/src/main/kotlin/io/dustin/apps/board/domain/restriction/posting/repository/PostingRestrictionRepository.kt
package io.dustin.apps.board.domain.restriction.posting.repository

import io.dustin.apps.board.domain.restriction.posting.model.PostingRestriction
import io.dustin.apps.common.repository.BaseRepository
import org.springframework.data.jpa.repository.Query
import java.util.*

interface PostingRestrictionRepository : BaseRepository<PostingRestriction, Long> {
    @Query("SELECT s FROM PostingRestriction s WHERE s.fromUserId = :fromUserId AND s.toUserId = :toUserId AND s.postingId = :postingId")
    fun findByUsers(
        fromUserId: Long,
        toUserId: Long,
        postingId: Long
    ): PostingRestriction?

    @Query("DELETE FROM PostingRestriction s WHERE s.fromUserId = :fromUserId AND s.toUserId = :toUserId AND s.postingId = :postingId")
    fun deleteByUsers(
        fromUserId: Long,
        toUserId: Long,
        postingId: Long
    )

}


